## Corporacion Favorita data set

This a public dataset from a Kaggle competition. More information at <a href='https://www.kaggle.com/c/favorita-grocery-sales-forecasting'>the competition site</a>. <br /> 
The dataset can be downloaded  by logging into Kaggle and using this <a href='https://www.kaggle.com/c/7391/download-all'> link </a>. 


In [2]:
import pandas as pd
import sys

In [3]:
df = pd.read_csv('train.csv',low_memory=False)

In [4]:
sys.getsizeof(df)

16471354832

Let's see what the data set is about

In [5]:
df.head(2)

id        date  store_nbr  item_nbr  unit_sales onpromotion
0   0  2013-01-01         25    103665         7.0         NaN
1   1  2013-01-01         25    105574         1.0         NaN

In [6]:
df['date'] = pd.to_datetime(df['date'])

In [7]:
df.shape[0]

125497040

In [8]:
df['store_nbr'].unique().shape[0]

54

In [9]:
df['item_nbr'].unique().shape[0]

4036

In [10]:
df['date'].min()

Timestamp('2013-01-01 00:00:00')

In [11]:
df['date'].max()

Timestamp('2017-08-15 00:00:00')

So we have registered over 125 million sales of 4036 different product across 54 different stores from 2013/01/01 to 2017/08/15.

As we are working with a really huge data set, for simplicity we prefer to use a sample of the Data Set, for example choosing a single store.

In [12]:
df[df['store_nbr'] == 25].to_csv('data/train_sample.csv')

## Next steps

In the <a href='2.ExploreAndFormat.ipynb' >next notebook </a> we are going to visualize and explore the time series from this sample and extract the columns that Amazon Forecast need as the model's input.